### PR TITLE
add tutorial notebooks (PR 7 of N)

### PR DESCRIPTION
**Pull request by bkmartinjr**
Add three notebooks (some are partial - functional but need additional verbiage before release).
* tutorial_pytorch: demonstrate toy model using a DataPipe
* tutorial_lightning: demonstrate toy model using Lighnting
* tutorial_multiworker: demo sharp edges with distributed/multi-worker usage (e.g., set_epoch)

### CODE REVIEW

File: notebooks/tutorial_pytorch.ipynb [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+    "\n",
+    "### `ExperimentAxisQueryIterDataPipe` parameters explained\n",
+    "\n",
+    "The constructor only requires a single parameter, `experiment`, which is a `soma.Experiment` containing the data of the organism to be used for training.\n",
+    "\n",
+    "To retrieve a subset of the Experiment's data, along either the `obs` or `var` axes, you may specify query filters via the `obs_query` and `var_query` parameters, which are both `soma.AxisQuery` objects.\n",
+    "\n",
+    "The values for the prediction label(s) that you intend to use for training are specified via the `obs_column_names` array.\n",
```
Comment (aaronwolen) [inline notebooks/tutorial_pytorch.ipynb:87]:
```suggestion
    "The values for the prediction label(s) that you intend to use for training are specified via the `obs_column_names` (or `var_column_names`) array.\n",
```

File: notebooks/tutorial_pytorch.ipynb [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+    "\n",
+    "The `batch_size` allows you to specify the number of obs rows (cells) to be returned by each return PyTorch tensor. You may exclude this parameter if you want single rows (`batch_size=1`).\n",
```
Comment (aaronwolen) [inline notebooks/tutorial_pytorch.ipynb:89]:
```suggestion
    "The `batch_size` parameter allows you to specify the number of `obs` rows (i.e., cells) to be returned by each return PyTorch tensor. You may exclude this parameter if you want single rows (`batch_size=1`).\n",
```

File: notebooks/tutorial_pytorch.ipynb [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+    "\n",
+    "The `shuffle` flag allows you to randomize the ordering of the training data for each training epoch. Note:\n",
+    "* You should use this flag instead of the `DataLoader` `shuffle` flag, primarily for performance reasons.\n",
+    "* PyTorch's TorchData library provides a [Shuffler](https://pytorch.org/data/main/generated/torchdata.datapipes.iter.Shuffler.html) `DataPipe`, which is alternate mechanism one can use to perform shuffling of an `IterableDataset`. However, the `Shuffler` will not \"globally\" randomize the training data, as it only \"locally\" randomizes the ordering of the training data within fixed-size \"windows\". Due to the layout of Census data, a given \"window\" of Census data may be highly homogeneous in terms of its `obs` axis attribute values, and so this shuffling strategy may not provide sufficient randomization for certain types of models."
```
Comment (aaronwolen) [inline notebooks/tutorial_pytorch.ipynb:93]:
```suggestion
    "* PyTorch's TorchData library provides a [Shuffler](https://pytorch.org/data/main/generated/torchdata.datapipes.iter.Shuffler.html) `DataPipe`, which is an alternate mechanism one can use to perform shuffling of an `IterableDataset`. However, the `Shuffler` will not \"globally\" randomize the training data, as it only \"locally\" randomizes the ordering of the training data within fixed-size \"windows\". This problematic for atlas-style datasets such as Census, where a given \"window\" of data may be highly homogeneous in terms of its `obs` axis attribute values, and so this shuffling strategy may not provide sufficient randomization for certain types of models."
```

File: notebooks/tutorial_lightning.ipynb [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+   "source": [
+    "# Training a model with PyTorch Lightning\n",
+    "\n",
+    "This tutorial provides a quick overview of training a toy model with Lightning, using the `tiledbsoma_ml.ExperimentAxisQueryIterableDataset` class, on data from the [CZI CELLxGENE Census](https://chanzuckerberg.github.io/cellxgene-census/) dataset. This is intended only to demonstrate the use of the `ExperimentAxisQueryIterableDataset`, and not as an example of how to train a biologically useful model.\n",
+    "\n",
+    "For more information on these API,  please refer to the [`tutorial_pytorch` notebook](tutorial_pytorch.ipynb).\n",
+    "\n",
+    "**Prerequesites**\n",
+    "\n",
+    "Install `tiledbsoma_ml` and `scikit-learn`, for example:\n",
```
Comment (aaronwolen) [inline notebooks/tutorial_lightning.ipynb:15]: Does `pytorch_lightning` need to be listed here too?

File: notebooks/tutorial_multiworker.ipynb [hunk 5]
```diff
@@ -0,0 +1,245 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Multi-process training\n",
+    "\n",
+    "Multi-process usage of `tiledbsoma_ml.ExperimentAxisQueryIterDataset` includes both:\n",
+    "* using the `torch.utils.data.DataLoader` with 1 or more worker (ie., with an argument of `n_workers=1` or greater)\n",
```
Comment (aaronwolen) [inline notebooks/tutorial_multiworker.ipynb:10]:
```suggestion
    "* using the `torch.utils.data.DataLoader` with 1 or more workers (i.e., with an argument of `n_workers=1` or greater)\n",
```

**Review by aaronwolen — APPROVED** [review]
The notebooks are really helpful and ran perfectly for me. I made a few minor comments/suggestions.

File: notebooks/tutorial_lightning.ipynb (hunk 4 above)
Comment (ryan-williams) [reply notebooks/tutorial_lightning.ipynb]: (added)

Comment (bkmartinjr) [reply notebooks/tutorial_lightning.ipynb]: Also fixed this in tutorial_pytorch.ipynb

**Pull request merged**